docs/schema-conversion-logic.md

```diff
@@ -0,0 +1,98 @@
+
+When the schema conversion logic needs to decrease the interval (increase resolution) of an archive with the aggregation method `average` and the destination archive is the raw/first archive, then it simply repeats the present datapoints to fill in the gaps. 
+
+For example when converting `2` datapoints from a `5s` interval archive into a `1s` interval archive, then it would look like this:
+
+```
+Input:
+  Avg: [1 2]
+Output:
+  Avg: [1 1 1 1 1 2 2 2 2 2]
+```
+
+However, if the destination archive for the same input is a rollup, so not the first archive of the schema, then it needs to generate a `sum` and a `count` aggregation because this is what Metrictank expects. Additionally it tries to ensure that the `sums` and `counts` are correct, as if they would have been generated from the first/raw archive. To do that it multiplies all the input points by the ratio of `input interval/output interval` and then duplicates the result `ratio` times, it also sets all the counts to `ratio`. This way the resulting averages stay correct, plus the counts and sums have values which are closer to what they would really be if the raw metric used as input had a value at every point without any gaps.
+
+```
+Input:
+  Avg: [1 2]
+Output:
+  Sum: [5 5 5 5 5 10 10 10 10 10]
+  Cnt: [5 5 5 5 5 5 5 5 5 5]
+```
+When viewed in Metrictank it divides the sums by the counts, so the averages stay correct.
+
+##### Last, Min, Max
+
+The cases of the `last`, `min` and `max` aggregation methods are more simple. When decreasing the interval, the point values simply get repeated to fill all the gaps. So for example when converting the following input datapoints which have been aggregated with `last` from a `1s` interval to a `2s` interval, it would look like this:
+
+```
+Input:
+  Lst: [1 2]
+Output:
+  Lst: [1 1 2 2]
+```
+
+#### Combining multiple input archives into one destination archive
+
+In some cases it is possible that multiple input archives need to be combined into one destination archive. This is because on one hand we are trying to achieve the highest resolution possible up to the resolution of the destination archive, while at the same time we are also trying to fill the whole retention period of the destination archive. This example shows what would happen if two input archives have to be combined into a single destination archive, to achieve both of the beforementioned goals.
+
+```
+Aggregation method: sum
+Destination archive: 1s:30s
+Input archives: 1s:5s,5s:30s
+Input Point Values:
+  Archive 1:
+    Sum: [1 2 3 4 5]
+  Archive 2:
+    Sum: [15 40 65 90 115 140]
```

Comment:
Maintenance might be a little obnoxious, but it make sense to align the arrays by adding some additional spacing?
```
  Archive 1:
    Sum: [1  2  3  4  5]
  Archive 2:
    Sum: [15 40 65 90 115 140]
```

Reply:
yeah good idea, i don't expect this will be edited often

Reply:
added formatting